brilift/src/main.rs

```diff
@@ -7,14 +7,62 @@ use cranelift_codegen::settings::Configurable;
 use cranelift_codegen::{ir, isa, settings};
 use cranelift_frontend::{FunctionBuilder, FunctionBuilderContext, Variable};
 use cranelift_jit::{JITBuilder, JITModule};
-use cranelift_module::{default_libcall_names, Module};
+use cranelift_module::{default_libcall_names, FuncOrDataId, Module};
 use cranelift_object::{ObjectBuilder, ObjectModule};
-use enum_map::{enum_map, Enum, EnumMap};
 use std::collections::HashMap;
 use std::fs;
 
-/// Runtime functions used by ordinary Bril instructions.
-#[derive(Debug, Enum)]
+trait RTFuncUtils {
+    fn sig(
+        &self,
+        pointer_type: ir::Type,
+        call_conv: cranelift_codegen::isa::CallConv,
+    ) -> ir::Signature;
+
+    fn name(&self) -> &'static str;
+
+    fn get_func_id<M: Module>(
+        &self,
+        module: &mut M,
+        call_conv: cranelift_codegen::isa::CallConv,
+    ) -> cranelift_module::FuncId {
+        match module.get_name(self.name()) {
+            Some(FuncOrDataId::Func(f)) => f,
+            Some(_) => panic!("Expected {} to be a function, not data", self.name()),
+            None => module
+                .declare_function(
```

Comment:
On the other hand, it's possible that putting this `declare_function` call into a utility function that takes a `&mut Module` or whatever might be a good idea for concision…